torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -5548,3 +5563,31 @@ def _test_rref_proxy_timeout(self, rref_proxy_api):
     def test_rref_proxy_timeout(self):
         for rpc_api in ["rpc_sync", "rpc_async", "remote"]:
             self._test_rref_proxy_timeout(rpc_api)
+
+    @skip_if_lt_x_gpu(1)
+    def test_rref_to_here_synchronization(self):
+        dst = worker_name((self.rank + 1) % self.world_size)
+        options = self.rpc_backend_options
+        options.set_device_map(dst, {0: 0})
```

Comment:
shall we add one more test for cross-device `to_here`? E.g., the forward output can resides on `cuda:1`, and `to_here()` fetch it back to `cuda:0`.

Reply:
added `test_rref_to_here_synchronization_cross_device`